From 915cca1ffb693625d2cabd80b434a24fd47d9c9a Mon Sep 17 00:00:00 2001
From: Kristoffer Flores <kflores@blackberry.com>
Date: Thu, 8 Aug 2013 13:59:54 -0400
Subject: [PATCH 20/61] Update blackberry.invoked unit test to fix race
 condition

In some cases success was not yet called when invocation handler completed.

Reviewed by Jeffrey Heifetz <jheifetz@blackberry.com>
---
 test/integration/automatic/blackberry.invoked.js |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/test/integration/automatic/blackberry.invoked.js b/test/integration/automatic/blackberry.invoked.js
index dbb8e50..ea2aa1f 100644
--- a/test/integration/automatic/blackberry.invoked.js
+++ b/test/integration/automatic/blackberry.invoked.js
@@ -19,7 +19,8 @@ describe("blackberry.invoked", function () {
     });
 
     it("should sucessfully send and receive invocations", function () {
-        var onSuccess = jasmine.createSpy(),
+        var successHandled = true,
+            onSuccess = jasmine.createSpy().andCallFake(function () { successHandled = true;}),
             onError = jasmine.createSpy(),
             invocationHandled = false,
             invokedTarget, invokedType, invokedAction, invokedData;
@@ -50,7 +51,7 @@ describe("blackberry.invoked", function () {
         });
 
         waitsFor(function () {
-            return invocationHandled;
+            return invocationHandled && successHandled;
         }, "invocation handler did not finish", 1000);
 
         runs(function () {
-- 
1.7.10.msysgit.1

